pkg/cloudprovider/types.go

```diff
@@ -35,12 +35,20 @@ type Capacity interface {
 	Create(context.Context, *Constraints) ([]Packing, error)
 	// Delete nodes in cloudprovider
 	Delete(context.Context, []*v1.Node) error
+	// GetInstanceTypes returns the instance types supported by the cloud provider.
+	GetInstanceTypes(ctx context.Context) ([]string, error)
+	// GetZones returns the zones supported by the cloud provider.
+	GetZones(ctx context.Context) ([]string, error)
+	// GetArchitectures returns the architectures supported by the cloud provider.
+	GetArchitectures(ctx context.Context) ([]string, error)
+	// GetOperatingSystems returns the operating systems supported by the cloud provider.
+	GetOperatingSystems(ctx context.Context) ([]string, error)
```

Comment:
is this just supporting linux / windows?  Is there a way to support non-bottlerocket since that's linux, but it's not AL2.

Reply:
MAC!